src/HassClient/Client/HassClient.cs

```diff
@@ -935,7 +935,7 @@ internal virtual Task SendMessage(HassMessageBase message, bool waitForResponse
             _logger.LogTrace($"Sends message {message.Type}");
             if (message is CommandMessage commandMessage)
             {
-                commandMessage.Id = ++_messageId;
+                commandMessage.Id = Interlocked.Increment(ref _messageId);

```

Comment:
Ohh nice find. That was not thread safe... even if the chances was slim it was making a difference it still more robust